logstash-core/src/main/java/org/logstash/KeyNode.java

```diff
@@ -60,8 +60,8 @@ private static String toString(Object value, String delim) {
         if (value == null) return "";
         if (value instanceof List) return join((List)value, delim);
         if (value instanceof BiValue) {
-            return ((BiValue) value).toString();
+            return value.toString();
```

Comment:
@jakelandis `BiValue#toString` is an override to `Object#toString` here, since we know that the `value` is a `BiValue` => `BiValue#toString` will be called (now that I'm explaining this I'm getting the urge to clean this up :P ... we should really not have chains of type checks ...)